CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+target_compile_definitions(s2nStatic PRIVATE -D_POSIX_C_SOURCE=200809L -D_FORTIFY_SOURCE=2)
+target_compile_options(s2nStatic PUBLIC -fPIC)    
+
+if(NO_STACK_PROTECTOR)
+    target_compile_options(s2nStatic PRIVATE -Wstack-protector -fstack-protector-all)
+
+    if(BUILD_SHARED_LIBS)
+        target_compile_options(s2nShared PRIVATE -Wstack-protector -fstack-protector-all)
+    endif()
+endif()
+
+if(S2N_UNSAFE_FUZZING_MODE)
+    target_compile_options(s2nStatic PRIVATE -fsanitize-coverage=trace-pc-guard -fsanitize=address,undefined,leak)
+
+    if(BUILD_SHARED_LIBS)
+        target_compile_options(s2nShared PRIVATE -fsanitize-coverage=trace-pc-guard -fsanitize=address,undefined,leak)
+    endif()
+endif()
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+
+find_package(LibCrypto REQUIRED)
+target_link_libraries(s2nStatic PRIVATE LibCrypto::Crypto PRIVATE ${OS_LIBS})
+
+if(BUILD_SHARED_LIBS)
+    target_link_libraries(s2nShared PRIVATE LibCrypto::Crypto PRIVATE ${OS_LIBS})
+endif()
+
+if(BUILD_SHARED_LIBS)
+    target_include_directories(s2nShared PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include>)
+    target_include_directories(s2nShared PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/api> $<INSTALL_INTERFACE:include>)
+endif()
+
+target_include_directories(s2nStatic PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include>)
+target_include_directories(s2nStatic PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/api> $<INSTALL_INTERFACE:include>)
+
+file(GLOB TESTLIB_SRC "tests/testlib/*.c")
+file(GLOB TESTLIB_HEADERS "tests/testlib/*.h")
+
+add_library(testss2n STATIC ${TESTLIB_HEADERS} ${TESTLIB_SRC})
+target_include_directories(testss2n PUBLIC tests)
+target_link_libraries(testss2n PUBLIC s2nStatic)
+
+#run unit tests
+file (GLOB TEST_LD_PRELOAD "tests/LD_PRELOAD/*.c")
+add_library(allocator_overrides SHARED ${TEST_LD_PRELOAD})
+
+include(CTest)
+enable_testing()
+
+file(GLOB UNITTESTS_SRC "tests/unit/*.c")
+    foreach(test_case ${UNITTESTS_SRC})
+    string(REGEX REPLACE ".+\\/(.+)\\.c" "\\1" test_case_name ${test_case})
+    add_executable(${test_case_name} ${test_case})
+    target_link_libraries(${test_case_name} PRIVATE testss2n PRIVATE m pthread)
+    target_include_directories(${test_case_name} PRIVATE api)
+    target_include_directories(${test_case_name} PRIVATE ./)
+    target_include_directories(${test_case_name} PRIVATE tests)
+    add_test(NAME ${test_case_name} COMMAND $<TARGET_FILE:${test_case_name}> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/unit)
+   
+    set_property(
+    TEST
+        ${test_case_name}
+    PROPERTY
+        ENVIRONMENT LD_PRELOAD=$<TARGET_FILE:allocator_overrides>)
+
+endforeach(test_case)
+
+set(export_targets s2nStatic)
+if(BUILD_SHARED_LIBS)
+    set(export_targets s2nStatic s2nShared)
+endif()
+
+#install the s2n files
+install (FILES ${API_HEADERS} DESTINATION "include/s2n/api")
```

Comment:
I think we should only install the api header. All of the other headers are internal currently.

Reply:
If we do that it's probably better to just install to include/s2n.h

Agreed?

Reply:
Sounds good to me